### PR TITLE
feat(staking): [LW-8879] persistence banner fix, minor banners refactor

### PR DESCRIPTION
**Pull request by refi93**
# Checklist

- [x] JIRA - LW-8879
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

Show the banner announcing persistence only if users are not persisting their portfolio yet. In the first commit, as a preparation, this PR is also consolidating the drift notification banner into the "StakingNotificationBanners" folder to keep related banners in a single place and extracts the helpers (including the one to determine if portfolio has been persisted) into the `store` folder, as suggested in a TODO comment (can be seen in the diff)

## Testing

To test the behavior you can manually delete the local storage keys(`multidelegationFirstVisit` and `multidelegationFirstVisitSincePortfolioPersistence`) in the developer console (Application -> Storage) and refresh the page - if user isn't multidelegating (with persistence) already the respective modal (multistaking vs persistence announcement) should show up. Scenarios

1. deleting only `multidelegationFirstVisitSincePortfolioPersistence` -> persistence announcement should show on wallet with no delegations
2. deleting both `multidelegationFirstVisit` and `multidelegationFirstVisitSincePortfolioPersistence` -> initial announcement should show for wallet with no delegations

Test also in popup mode - same banners should be visible (after clearing the localstorage keys)

Test

## Screenshots
Screenshot 2023-11-07 at 09 05 15
Screenshot 2023-11-07 at 09 14 25

### CODE REVIEW

File: apps/browser-extension-wallet/src/views/browser-view/features/staking/components/MultiDelegationStaking.tsx [hunk 1]
```diff
@@ -109,10 +109,15 @@ export const MultiDelegationStaking = (): JSX.Element => {
         // TODO: LW-7575 make compactNumber reusable and not pass it here.
         compactNumber: compactNumberWithUnit,
         multidelegationFirstVisit,
-        triggerMultidelegationFirstVisit: () => setMultidelegationFirstVisit(false),
+        triggerMultidelegationFirstVisit: () => {
+          setMultidelegationFirstVisit(false);
```
Comment (refi93) [inline apps/browser-extension-wallet/src/views/browser-view/features/staking/components/MultiDelegationStaking.tsx:113]: this is important to ensure users don't get to see the "persistence" banner after seeing the "first visit" banner and vice-versa, which isn't desired

File: packages/staking/src/features/staking/OneTimeModals.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { useDelegationPortfolioStore } from 'features/store';
+import { isPortfolioSavedOnChain } from 'features/store/delegationPortfolioStore/isPortfolioSavedOnChain';
```
Comment (refi93) [inline packages/staking/src/features/staking/OneTimeModals.tsx:2]: I wasn't able to export this helper from `features/store/delegationPortfolioStore/index.ts` and I'm not sure exactly why TBH. The build just failed with this cryptic error and the only way around I found was to import directly:

```
E/STAKING] CLI Cleaning output folder
 | [LACE/STAKING] ESM Build start
 | [LACE/STAKING] CJS Build start
 | ✘ [ERROR] No matching export in "src/features/store/index.ts" for import "isPortfolioSavedOnChain"
 | 
 |     src/features/staking/OneTimeModals.tsx:1:9:
 |       1 │ import { isPortfolioSavedOnChain, useDelegationPortfolioStore } fro...
 |         ╵          ~~~~~~~~~~~~~~~~~~~~~~~
 | 
 | [LACE/STAKING] ESM Build failed
 | Error: Build failed with 1 error:
 | src/features/staking/OneTimeModals.tsx:1:9: ERROR: No matching export in "src/features/store/index.ts" for import "isPortfolioSavedOnChain"
 |     at failureErrorWithLog (/Users/rafaelkorbas/workplace/lace/node_modules/esbuild/lib/main.js:1636:15)
 |     at /Users/rafaelkorbas/workplace/lace/node_modules/esbuild/lib/main.js:1048:25
 |     at runOnEndCallbacks (/Users/rafaelkorbas/workplace/lace/node_modules/esbuild/lib/main.js:1471:45)
 |     at buildResponseToResult (/Users/rafaelkorbas/workplace/lace/node_modules/esbuild/lib/main.js:1046:7)
 |     at /Users/rafaelkorbas/workplace/lace/node_modules/esbuild/lib/main.js:1075:16
 |     at responseCallbacks.<computed> (/Users/rafaelkorbas/workplace/lace/node_modules/esbuild/lib/main.js:697:9)
 |     at handleIncomingPacket (/Users/rafaelkorbas/workplace/lace/node_modules/esbuild/lib/main.js:752:9)
 |     at Socket.readFromStdout (/Users/rafaelkorbas/workplace/lace/node_modules/esbuild/lib/main.js:673:7)
 |     at Socket.emit (node:events:513:28)
 |     at addChunk (node:internal/streams/readable:324:12)
 |     at readableAddChunk (node:internal/streams/readable:297:9)
 |     at Readable.push (node:internal/streams/readable:234:10)
 |     at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
 | ✘ [ERROR] No matching export in "src/features/store/index.ts" for import "isPortfolioSavedOnChain"
 | 
 |     src/features/staking/OneTimeModals.tsx:1:9:
 |       1 │ import { isPortfolioSavedOnChain, useDelegationPortfolioStore } fro...
 |         ╵          ~~~~~~~~~~~~~~~~~~~~~~~
 | 
 | [LACE/STAKING] CJS Build failed
 | [LACE/STAKING] DTS ⚡️ Build success in 4810ms
@lace/browser-extension-wallet
 | <s> [webpack.Progress] 3% setup watch run
 | <s> [webpack.Progress] 3% setup watch run webpack-cli
 | <s> [webpack.Progress] 3% setup watch run ForkTsCheckerWebpackPlugin
```
```

File: packages/staking/src/features/staking/OneTimeModals.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    return (
+      <MultidelegationBetaModal
+        visible={multidelegationFirstVisit}
+        onConfirm={triggerMultidelegationFirstVisit}
+        popupView={popupView}
+      />
+    );
+  }
+
+  if (!portfolioSavedOnChain) {
```
Comment (refi93) [inline packages/staking/src/features/staking/OneTimeModals.tsx:49]: originally, this modal wasn't included in the popup view but I believe this was just an omission given the original task doesn't specify that limitation, so I added it there, too: https://input-output.atlassian.net/browse/LW-8837

Comment (refi93) [reply packages/staking/src/features/staking/OneTimeModals.tsx:49]: I just asked Fernando to confirm, waiting for reply

File: packages/staking/src/features/staking/OneTimeModals.tsx (hunk 2 above)
Comment (xdzurman) [reply packages/staking/src/features/staking/OneTimeModals.tsx:2]: This happened also to Szymon and me, we didn't manage to fix it in a short time, not worth inquiring right now :/

**Review by xdzurman — APPROVED** [review]

**Review by mrcnk — APPROVED** [review]

File: packages/staking/src/features/staking/OneTimeModals.tsx (hunk 3 above)
Comment (refi93) [reply packages/staking/src/features/staking/OneTimeModals.tsx:49]: confirmed with Fernando that we also want the "persistence" modal in the popup view 👍

**Review by oldGreg5 — APPROVED** [review]

**Pull request merged**